src/vos/vos_io.c

```diff
@@ -1252,26 +1261,66 @@ vos_update_end(daos_handle_t ioh, uint32_t pm_ver, daos_key_t *dkey, int err)
 	err = vos_publish_blocks(ioc->ic_obj, &ioc->ic_blk_exts, true);
 
 abort:
-	err = err ? umem_tx_abort(umem, err) : umem_tx_commit(umem);
+	if (err == -DER_INPROGRESS && dth != NULL &&
+	    (dth->dth_flags & DTX_F_AOC) &&
+	    !daos_is_zero_dti(&dth->dth_conflict)) {
+		int	rc;
+
+		/* Abort the conflict DTX by force, then try again. */
```

Comment:
Could you explain it more?

Reply:
It is possible that the non-leader rep failed to abort some DTX, but leader has already aborted. Under such case, the leader will execute current modification successfully, but the current non-leader replica failed to do that because conflict with such bad DTX that should have been aborted. Under such case, we should abort such DTX by force. I will add more comments in the patch.